### PR TITLE
Try re-enabling CancellationToken tests in a loop

### PR DESCRIPTION
**Pull request by jkotalik**
To diagnose https://github.com/aspnet/AspNetCore/issues/4621.

### CODE REVIEW

**Comment by jkotalik** [comment]
@Tratcher @davidfowl I've tried multiple things to try to reproduce this issue:
- Go onto the CI machine that had the failure and run all four fail tests ~50000 times. 
- Run with my computer on low power mode. 
- Add Task.Delays in appropriate places

I added some logging (which is bad for perf) to try to catch the issue when it occurs. Other than that, I'm open to anything.

File: src/Http/Http/src/Microsoft.AspNetCore.Http.csproj [hunk 1]
```diff
@@ -18,6 +18,7 @@
     <Reference Include="Microsoft.AspNetCore.WebUtilities" />
     <Reference Include="Microsoft.Extensions.ObjectPool" />
     <Reference Include="Microsoft.Extensions.Options" />
+    <Reference Include="Microsoft.Extensions.Logging" />
```
Comment (Tratcher) [inline src/Http/Http/src/Microsoft.AspNetCore.Http.csproj]: I hesitate to add this only for the pipe components that are going to move later. How will we remember to remove this?

Comment (jkotalik) [reply src/Http/Http/src/Microsoft.AspNetCore.Http.csproj]: Pretty simple, once we move the types won’t exist :)

Comment (Tratcher) [reply src/Http/Http/src/Microsoft.AspNetCore.Http.csproj]: How do we remember to remove the logging dependency?

Comment (jkotalik) [reply src/Http/Http/src/Microsoft.AspNetCore.Http.csproj]: I mean all Microsoft.Extensions/ Microsoft.AspNetCore dependencies would be removed/not used.

**Comment by jkotalik** [comment]
@Tratcher I'm just going to re-enable the tests here. I couldn't repro by running this on aspnetci.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**